### PR TITLE
MDEV-19208: Pkgconf fixes

### PR DESCRIPTION
**Pull request by mathstuf**

### CODE REVIEW

**Comment by vuvova** [comment]
No, don't worry. We'll rebase if needed.

File: debian/libmariadb-dev-compat.links [hunk 1]
```diff
@@ -1,3 +1,3 @@
 usr/bin/mariadb_config usr/bin/mysql_config
 usr/include/mariadb usr/include/mysql
-usr/share/pkgconfig/mariadb.pc usr/share/pkgconfig/mysqlclient.pc
+usr/lib/*/pkgconfig/mariadb.pc usr/lib/*/pkgconfig/mysqlclient.pc
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply debian/libmariadb-dev-compat.links]: relevant policy: https://www.debian.org/doc/debian-policy/ch-sharedlibs.html#shared-library-support-files and https://wiki.ubuntu.com/MultiarchCross#line-92

[the remark this answers is not in the document]
Comment (mathstuf) [reply debian/libmariadb-dev-compat.links]: Looking at glib's Debian package, it looks like there is `${DEB_HOST_MULTIARCH}` for this. Not sure though.

**Comment by grooverdan** [comment]
I submitted #597 a while ago addressing multiarch. Thought this sounded familiar.

**Comment by grooverdan** [comment]
From https://wiki.debian.org/Multiarch/Implementation#CMake, `include(GNUInstallDirs)` and use `CMAKE_INSTALL_LIBDIR` as `INSTALL_LIBDIR` might be an option.

**Comment by mathstuf** [comment]
@grooverdan Agreed, but that's a larger change that should probably be done with another PR.

**Comment by an3l** [comment]
Hi @mathstuf ,

Thanks for your contribution.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license.

The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/

Please indicate in a comment below that you are contributing your new code of the whole pull request, including one or several files that are either new files or modified ones, under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
Anel

**Comment by mathstuf** [comment]
BSD-new is OK with me for this entire PR.

**Comment by mathstuf** [comment]
Ping?

**Comment by mathstuf** [comment]
Seems that it is not the right way. https://www.debian.org/doc/manuals/maint-guide/advanced.en.html

```
Generate files like debian/foo.links from debian/foo.links.in dynamically by adding a script to the override_dh_auto_configure target in debian/rules.

override_dh_auto_configure:
        dh_auto_configure
        sed 's/@DEB_HOST_MULTIARCH@/$(DEB_HOST_MULTIARCH)/g' \
                debian/foo.links.in > debian/foo.links
```

**Comment by mathstuf** [comment]
Trying my update in a local build inside a container.

**Comment by mathstuf** [comment]
With the new version:

```
root@9a992016a014:/# dpkg -c libmariadb-dev-compat_10.4.0_amd64.deb 
drwxr-xr-x root/root         0 2018-08-05 05:06 ./
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/bin/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/include/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/lib/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/lib/x86_64-linux-gnu/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/lib/x86_64-linux-gnu/pkgconfig/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/share/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/share/doc/
drwxr-xr-x root/root         0 2018-08-05 05:06 ./usr/share/doc/libmariadb-dev-compat/
-rw-r--r-- root/root       511 2018-08-05 05:06 ./usr/share/doc/libmariadb-dev-compat/changelog.gz
-rw-r--r-- root/root      2502 2018-08-05 05:06 ./usr/share/doc/libmariadb-dev-compat/copyright
lrwxrwxrwx root/root         0 2018-08-05 05:06 ./usr/bin/mysql_config -> mariadb_config
lrwxrwxrwx root/root         0 2018-08-05 05:06 ./usr/include/mysql -> mariadb
lrwxrwxrwx root/root         0 2018-08-05 05:06 ./usr/lib/x86_64-linux-gnu/libmysqlclient.a -> libmariadbclient.a
lrwxrwxrwx root/root         0 2018-08-05 05:06 ./usr/lib/x86_64-linux-gnu/libmysqlclient.so -> libmariadb.so.3
lrwxrwxrwx root/root         0 2018-08-05 05:06 ./usr/lib/x86_64-linux-gnu/libmysqlclient_r.a -> libmariadbclient.a
lrwxrwxrwx root/root         0 2018-08-05 05:06 ./usr/lib/x86_64-linux-gnu/libmysqlclient_r.so -> libmariadb.so.3
lrwxrwxrwx root/root         0 2018-08-05 05:06 ./usr/lib/x86_64-linux-gnu/pkgconfig/mysqlclient.pc -> mariadb.pc
```

**Comment by an3l** [comment]
@ottok this patch is also similar to PR 597
What is your opinion about?

**Comment by grooverdan** [comment]
Preference was for #597 per MDEV-14340. It doesn't include mysql_config changes but with pkgconfig that's not needed. No point creating a backwards compatible thing for a use case that doesn't exist.

**Comment by mathstuf** [comment]
Even if #597 is merged, there are other changes here that are worthwhile (namely the `-I` change). I'm fine with rebasing on whatever ends up being merged.

File: support-files/mariadb.pc.in [hunk 2]
```diff
@@ -20,4 +20,4 @@ Description: @CPACK_PACKAGE_DESCRIPTION_SUMMARY@
 URL: @CPACK_PACKAGE_URL@
 Version: @VERSION@
 Libs: -L${libdir} @LIBS_FOR_CLIENTS@
-Cflags: -I${includedir} -I${includedir}/.. @CFLAGS_FOR_CLIENTS@
```
Comment (ottok) [inline support-files/mariadb.pc.in]: What does this line do?

**Comment by ottok** [comment]
I dislike the use of `.in` files in the `debian/` path. I once cleaned them away from the sources and don't want to have that practice re-introduced. It breaks all static analysis of Debian packaging files and in general the declarative nature of those files.

File: support-files/mariadb.pc.in (hunk 2 above)
Comment (mathstuf) [reply support-files/mariadb.pc.in]: I removed the `-I${includedir}/..` line which is not necessary (and adding `/usr/include` to the explicit include line messed up `#include_next` for some stdlib headers).

**Comment by mathstuf** [comment]
I'm just following the recommended practices documented by Debian for that. The other PR may have a just-as-suitable solution for it, but that's not what I saw in the docs/wiki.

**Comment by vaintroub** [comment]
I removed myself from the list of reviewers. I've no clue about this patch really, it is pkgconfig, and Debian , and all stuff I do not know.

**Comment by cvicentiu** [comment]
Hi @mathstuf !

This PR will be merged with a few changes.

First part is merged in 10.1:
f21592c675e198ba6298748b40e5d695a17f4e5f

Second part is c5c1027c6eb0db9738c231731b664f9736970d98. I accidentally merged the commit with me as the author. Sadly I can only give you credit for it here.

The changes to Debian packaging will be updated in 10.3 and 10.4 as necessary, but without the sed changes. There is a better option with dh-exec.

**Comment by mathstuf** [comment]
Thanks!

**Comment by cvicentiu** [comment]
And the Debian changes have landed in 10.3, with commit: fd2dc9c3fdfb73fa450db9fecd72044bb0554040

Thank you for your patience with having this issue resolved. It ended up being more involved than I anticipated, testing all possible configurations, as well as comparing to downstream debian / red-hat packaging.

**Pull request closed without merging**